test/phoenix/endpoint/instrument_test.exs

```diff
@@ -95,4 +99,21 @@ defmodule Phoenix.Endpoint.InstrumentTest do
     assert_receive {__MODULE__.MyInstrumenter, :common_event_stop}
     assert_receive {__MODULE__.MyOtherInstrumenter, :common_event_stop}
   end
+
+  test "event callbacks that raise/throw" do
+    import Endpoint
+
+    log = RouterHelper.capture_log fn ->
+      :ok = instrument :raising_event, fn ->
+        send self(), :ok
+      end
+    end
+
+    assert_receive :ok
+    # We have the correct stacktrace.
+    assert log =~ Path.relative_to_cwd(__ENV__.file)
+    # And we're sure the exception is logged twice:
+    err = "** (RuntimeError) oops"
+    assert Regex.scan(~r/#{Regex.escape(err)}/, log) == [[err], [err]]
```

Comment:
Maybe to be uber-pedantic we could use `^` (beginning of the line) and `$` (end of the line) in the regex? :bowtie: